### DISCUSSION TITLE
Firebase auth sharing between iOS and WatchOS app

### DISCUSSION THREAD

**Discussion by RohitRajendran** [firebase/firebase-ios-sdk#7840] [Q&A]
I'm working on an app where I'm using anonymous sign in from the iOS app and I would like to share that authentication with the WatchOS app. I've set the keychain access group to be the same on both the ios and watchos apps and call `Auth.auth().useUserAccessGroup` but the watchOS app returns nil for the user.

I came across some [old forum posts](https://developer.apple.com/forums/thread/5938) saying that keychain access sharing between iOS and Watch is no longer possible from WatchOS 2 but I wasn't sure if that was still the case.

Am I just doing something wrong or do I need to do something else to share authentication between the two?

**Comment by ncooke3** [comment]
Tracking in issue: https://github.com/firebase/firebase-ios-sdk/issues/7854

**Reply by dweekly** [reply]
That ticket was closed unanswered and locked. It would be nice to have some documentation that is up-to-date (>= watchOS 2) that shows how to sync auth state between iOS (15+) and watchOS (8+). The existing docs walk through sharing with app groups but this doesn't work on modern watchOS since the watch itself is running separately from the phone and can't access app group data.

I'd guess the right answer is to use keychain sharing along with iCloud sync but since this is a common use case it would be nice to have it spelled out a little more than on https://firebase.google.com/docs/auth/ios/single-sign-on

**Reply by dweekly** [reply]
It looks like yet others are also having trouble understanding best practices on synchronizing user login state between iOS and watchOS: https://github.com/firebase/firebase-ios-sdk/issues/9525

**Reply by paulb777** [reply]
Thanks for the comments. I reopened #7854

**Reply by lewis-smith** [reply]
Just to say 7854 was closed because it is not related to this. 

It would still be good to have some documentation on how to sync authentication to watchOS and what is and isn't possible.

https://firebase.google.com/docs/ios/learn-more#firebase_library_support_by_platform Suggests support is "partial". Without further documentation this feels less helpful than just saying it's unavailable.
